### DISCUSSION TITLE
Open large file

### DISCUSSION THREAD

**Discussion by applePrincess** [community/community#3373] [Mobile]
Open a large file fails with a message "Something went wrong", which makes sense but I want it to show something like 
Large file cannot view

And a button to download which could download the file in Download directory or view (kind of implies to download) the file in raw(preferably syntax highlighted).

**Comment by PranamBhat** [comment]
For large file - It's preferably GitHub is correct. On other pages as well it displays same thing if something went wrong. Means user should be notified with this alert!

For Download - it is a GitHub way to showcase the file in "raw" rather than direct download button - save. If GitHub makes everything easy over file download then it might be a loophole and users will be coming here to download and take every time. They should not make this as so easy.

**Reply by applePrincess** [reply]
> If GitHub makes everything easy over file download then it might be a loophole

I was wondering about this, but github webpage allows us to "download" file directly so I don't think it adds a new loophole. But yes, github app should be for browsing repo/source code and managing issues thus direct download may not be appropriate.

**Reply by PranamBhat** [reply]
Yes. Correct :)
